docs/modules/ROOT/pages/documentation/extensions/grpc/README.adoc

```diff
@@ -0,0 +1,415 @@
+                    </goals>
+                </execution>
+            </executions>
+        </plugin>
+    </plugins>
+</build>
+----
+
+* The `os-maven-plugin` extension is used to identify OS properties used during generation of stubs classes.
+* The `protobuf-maven-plugin` plugin calls the `protobuf` compiler (`protoc`) to generate the stubs classes.
+
+[[build-service]]
+=== Build the Service
+
+After doing the previous steps, you can build the project to generate the stubs. These stubs are needed to resolve the types used in the service implementation.
+
+Example:
+
+[source, shell]
+----
+${webapp root folder} > mvn clean install
+----
+
+The stubs should be generated with similar structure like the following:
+
+image::grpc/grpc-stubs-folders.png[Stubs folders]
+
+target/generated-sources/protobuf/grpc-java/fish.payara.samples.grpc:: Here you can see the service class
+target/generated-sources/protobuf/java/fish.payara.samples.grpc:: Here you can see the types associated to the service
+
+[[extend-service-implementation]]
+=== Extending service implementation class
+
+To implement the service's endpoint you'll have to extend the implementation base class that is nested inside the recently compiled gRPC stub class located under `/target/generated-sources/protobuf/grpc-java`:
+
+[source, Java]
+----
+@Dependent
+public class PayaraService extends PayaraServiceGrpc.PayaraServiceImplBase {
+    private final static Logger log = Logger.getLogger(PayaraService.class.getName());
+    @Override
+    public void communicate(fish.payara.samples.grpc.PayaraReq request,
+                            io.grpc.stub.StreamObserver<fish.payara.samples.grpc.PayaraResp> responseObserver) {
+        final String message = request.getMessage(); //getting message from the request
+        log.info(String.format("Processing message: %s", message)); //printing incoming message from the request
+        responseObserver.onNext(response(message)); //setting the message to the response
+        responseObserver.onCompleted(); //indicating that the response is complete
+    }
+
+    private static final fish.payara.samples.grpc.PayaraResp response(String message) {
+        return fish.payara.samples.grpc.PayaraResp.newBuilder() //creating builder
+                .setMessage(message) //setting response message
+                .build(); //build the response
+    }
+}
+----
+
+Our example is a simple "echo" service that will print a "Hello World" message. We can see that the  `communicate` method receives the `PayaraReq` and the `StreamObserver<fish.payara.samples.grpc.PayaraResp>` parameters which are needed to process the incoming request with the 'Hello World' message and create the response using the same message.
+
+[[deploy-service-application]]
+=== Deploy service application
+
+The final step is to deploy the application in a Payara Server domain. We can do this by using the Admin Console or the Asadmin CLI:
```

Comment:
An error has occurredError occurred during deployment: Exception while deploying the app [payara-web]. Please see server.log for more details.

Details:
Caused by: org.xml.sax.SAXParseException; lineNumber: 4; columnNumber: 17; Deployment descriptor file WEB-INF/web.xml in archive [payara-web].  O tipo de elemento "payara-web-app" deve ser declarado.
	at com.sun.enterprise.deployment.io.DeploymentDescriptorFile.read(DeploymentDescriptorFile.java:322)
	at com.sun.enterprise.deployment.io.DeploymentDescriptorFile.read(DeploymentDescriptorFile.java:239)
	at com.sun.enterprise.deployment.archivist.Archivist.readStandardDeploymentDescriptor(Archivist.java:713)
	at com.sun.enterprise.deployment.archivist.Archivist.readDeploymentDescriptors(Archivist.java:390)
	at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:271)
	at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:280)
	at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:240)
	at com.sun.enterprise.deployment.archivist.ApplicationFactory.openArchive(ApplicationFactory.java:161)
	at org.glassfish.javaee.core.deployment.DolProvider.processDOL(DolProvider.java:221)
	... 68 more
Caused by: org.xml.sax.SAXParseException; lineNumber: 4; columnNumber: 17; O tipo de elemento "payara-web-app" deve ser declarado.
...
[2022-03-11T10:04:08.873-0300] [Payara 5.2021.9] [INFORMAÇÕES] [] [org.glassfish.admingui] [tid: _ThreadID=421 _ThreadName=admin-thread-pool::admin-listener(10)] [timeMillis: 1647003848873] [levelValue: 800] [[
  Exception Occurred :Error occurred during deployment: Exception while deploying the app [payara-web] : org.xml.sax.SAXParseException; lineNumber: 4; columnNumber: 17; Deployment descriptor file WEB-INF/web.xml in archive [payara-web].  O tipo de elemento "payara-web-app" deve ser declarado.. Please see server.log for more details.]]
...
[2022-03-11T10:06:13.555-0300] [Payara 5.2021.9] [GRAVE] [NCLS-CORE-00026] [javax.enterprise.system.core] [tid: _ThreadID=418 _ThreadName=admin-thread-pool::admin-listener(8)] [timeMillis: 1647003973555] [levelValue: 1000] [[
  Exception during lifecycle processing
java.lang.NullPointerException
	at org.glassfish.web.deployment.node.runtime.gf.WebBundleRuntimeNode.getSunDescriptor(WebBundleRuntimeNode.java:214)

Reply:
this is caused because bad schema from the web.xml included on the project. This should need to be removed and it is only needed the payara-web.xml into the WEB-INF folder to configure the base-root